RFC_PROXY_AGENT.md

```diff
@@ -0,0 +1,302 @@
+# Public Cluster verification problems
+
+## Original Problems/Purpose
+
+In the original design of the Spore Protocol, if one wants to create a spore with a Cluster ID, then it must follow:
+
+1. The referenced Cluster Cell must be found in the `CellDeps`.
+2. The referenced Cluster Cell must be found in the `Inputs`.
+3. The referenced Cluster Cell must be found in the `Outputs`.
+
+This is for verifying the ownership of the referenced Cluster Cell, avoiding vicious behavious happened to one’s private Cluster.
+
+While this guarantees unexpected minting will not happen to a certain Cluster, it also brings a problem for Clusters that support public minting — those using specific lock to achieve this. That is, there might be a  bottleneck when the Cluster contains many popular Spore cells and there's only one Cluster cell available to construct transactions.
+
+More specificly, if two person wants to mint different Spores into a Cluster named `Cluster A`, and they accidently send their mint transaction at the same time, then one of their transaction will be rejected because of the Cluster was consumed and recreated, the rejected one also needs to reconstruct its transaction in order to continue minting.
+
+## Solution details
+
+### Step1: Creating Cluster Proxy Cell
+
+In this stage, the **owner** needs to create a special cell, here we called it **Cluster Proxy Cell**
+
+A Cluster Proxy Cell’s structure is like below:
+
+```yaml
+Cluster Proxy Cell:
+    Data: REFERENCED_CLUSTER_ID
+    Type:
+        code_hash: CLUSTER_PROXY_TYPE_HASH
+	      args: <cluster_proxy_id> [<min_payment>]
+    Lock:
+        <user_defined>
+```
+
+The Type args can be:
+
+- args: <cluster_proxy_id>
+- args: <cluster_proxy_id> <minimal payment in 10^n ckybytes: uint16>
```

Comment:
1. What is `ckybytes`?
2. @ShookLyngs said that type `uint16` should be `uint8`, right?